src/AccessControl/Login.php

```diff
@@ -83,15 +83,44 @@ public function login($userName, $password, AccessControlEvent $event)
     }
 
     /**
-     * Check a user login request for username/password combinations.
+     * Attempt to login a user with the given password. Accepts username or
+     * email.
      *
      * @param string             $userName
-     * @param string             $password
      * @param AccessControlEvent $event
      *
+     * @throws AccessControlException
+     *
      * @return bool
      */
-    protected function loginCheckPassword($userName, $password, AccessControlEvent $event)
+    public function loginAsUser($userName, AccessControlEvent $event)
+    {
+        $authCookie = $this->requestStack->getCurrentRequest()->cookies->get($this->authTokenName);
+
+        // Remove expired tokens
+        $this->getRepositoryAuthtoken()->deleteExpiredTokens();
+
+        if ($userName !== null) {
+            return $this->loginCheckUser($userName, null, $event, true);
+        } elseif ($authCookie !== null) {
+            return $this->loginCheckAuthtoken($authCookie, $event);
+        }
+
+        $this->systemLogger->error('Login function called with empty username/password combination, or no authentication token.', ['event' => 'security']);
```

Comment:
Message needs to be changed so it's clear it's from `loginAsUser`, as opposed to a login with password.

Reply:
like this?